src/jit/lsra.cpp

```diff
@@ -6653,23 +6656,25 @@ LinearScan::allocateRegisters()
                 }
                 else
 #endif // FEATURE_SIMD
-                if (currentRefPosition->IsActualRef())
+                if (currentRefPosition->RequiresRegister() ||
+                    currentRefPosition->IsRegOptional())
                 {
+                    // Though Lower/Codegen has indicated that it can generate code even if
+                    // no reg is allocated to this ref position, we will make an attempt
+                    // to get a busy reg if it is allocated to a lesser important ref position.
```

Comment:
This comment is confusing, because this then clause will be executed in the case where `RequiresRegister()` is true. It should start with something like "Even if this is marked as optional, we will make an attempt ..."
Also, in looking at this I think it needs to be restructured a bit. You've already done some analysis to determine whether we should try to allocate a register, with the setting of `allocateReg`. In the current structure, I believe that `allocateReg` is only true if the above condition is true. So I think you only need `if (allocateReg)`. Then I think the comment about making a second attempt belongs in the setting of `allocateReg` above.